openedx/features/partners/give2asia/tests/test_views.py

```diff
@@ -0,0 +1,174 @@
+from ddt import data, ddt
+from django.contrib.auth.models import User
+from django.core.urlresolvers import reverse
+
+from lms.djangoapps.onboarding.models import Organization, UserExtendedProfile
+from openedx.core.lib.api.test_utils import ApiTestCase
+from openedx.features.partners.tests.factories import FocusAreaFactory, OrganizationFactory, PartnerFactory
+
+
+@ddt
+class G2ARegistrationViewTest(ApiTestCase):
+    """
+    Includes test cases for g2a registration
+    """
+
+    NAME = 'bob james'
+    COUNTRY = 'Pakistan'
+    ORGANIZATION = 'arbisoft'
+    EMAIL = 'bob@example.com'
+    USERNAME = 'bob'
+    PASSWORD = 'Test@12345'
+    TERMS_OF_SERVICE = True
+
+    PARTNER = 'give2asia'
+
+    def setUp(self):
+        self.partner = PartnerFactory.create(slug=self.PARTNER, label=self.ORGANIZATION)
+        self.registration_url = reverse('partner_register', args=[self.PARTNER])
+
+    def test_put_not_allowed(self):
+        """
+        API should not accept http put method
+        :return : None
+        """
+        response = self.client.put(self.registration_url)
+        self.assertHttpMethodNotAllowed(response)
+
+    def test_delete_not_allowed(self):
+        """
+        API should not accept http delete method
+        :return : None
+        """
+        response = self.client.delete(self.registration_url)
+        self.assertHttpMethodNotAllowed(response)
+
+    def test_create_new_partner_user_with_organization(self):
+        """
+        Test user is not first learner if organization already exists
+        Create an organization and then try to register.
+        :return : None
+        """
+        OrganizationFactory(label=self.ORGANIZATION)
+
+        response = self.client.post(self.registration_url, {
+            'name': self.NAME,
+            'organization_name': self.ORGANIZATION,
+            'country': self.COUNTRY,
+            'email': self.EMAIL,
+            'username': self.USERNAME,
+            'password': self.PASSWORD,
+            'terms_of_service': self.TERMS_OF_SERVICE
+        })
+        self.assertHttpOK(response)
+
+        # get inserted data from the database
+        user = User.objects.filter(username=self.USERNAME).first()
```

Comment:
actually the code will crash if user isn't successfully added into the database.Thats why I put filter there.So, if user isn't registered it will return None instead of exception.